### PR TITLE
fix(portal): UX improvements

### PR DESCRIPTION
**Pull request by AndrewDryga**
This PR accumulates lots of small UX fixes from #6645.

### CODE REVIEW

**Review by jamilbk — COMMENTED** [review]
Took a glance through, looks good. Will defer approval to @bmanifold !

File: elixir/apps/web/lib/web/components/navigation_components.ex [hunk 1]
```diff
@@ -347,4 +347,25 @@ defmodule Web.NavigationComponents do
...
+    <.website_link href="/kb/deploy/gateways" class="text-neutral-900">Deploy Gateway(s)</.website_link>
+    <.website_link href={~p"/contact/sales"}>Contact Sales</.website_link>
+  """
+  attr :path, :string, required: true
+  attr :fragment, :string, required: false, default: ""
+  attr :rest, :global
+
+  def docs_action(assigns) do
+    ~H"""
+    <.link navigate={"/kb#{@path}?utm_source=product##{@fragment}"} target="_blank" {@rest}>
```
Comment (jamilbk) [inline elixir/apps/web/lib/web/components/navigation_components.ex]: Causes broken links

```suggestion
    <.link href={"https://www.firezone.dev/kb#{@path}?utm_source=product##{@fragment}"} target="_blank" {@rest}>
```

Comment (jamilbk) [reply elixir/apps/web/lib/web/components/navigation_components.ex]: Also, `target="_blank"` has no effect here, phoenix just redirects instead.

**Review by jamilbk — APPROVED** [review]
Fixed the link issues and adjusted a couple other minor bugs with input placeholder overflow

Screenshot 2024-10-12 at 4 18 40 PM

**Comment by jamilbk** [comment]
@bmanifold Made some small UX tweaks to the new DNS form since it was a merge conflict with this PR, hope you don't mind:

- Use `Upstream resolver` terminology consistently throughout the page
- Show helptext for form empty state

**Comment by jamilbk** [comment]
@AndrewDryga I tried to fix the tests, but there's still one that's causing issues. There seems to be bugs with the auto-complete logic on the sign up form. If you enter an email, then delete it, the `Company Name` contains residual input:

Screenshot 2024-10-12 at 6 20 36 PM


Do we need all this autocomplete logic in this form? I don't think anyone has complained that about the friction around our sign up process or having to fill these in specifically, also I wonder how many people will accept the defaults vs change them. If more people are changing them (like we see with the Site name during site creation), then people have to read the text, then make the decision to delete it, vs just writing the company name on their own.

In other words, this seems optimized for starter plan UX, where most business users will need to delete this and use their org name here. Is that what we want?

File: elixir/apps/web/lib/web/live/settings/dns.ex [hunk 2]
```diff
@@ -32,24 +32,28 @@ defmodule Web.Settings.DNS do
...
           <h2 class="mb-4 text-xl text-neutral-900">Client DNS</h2>
-          <p class="mb-4 text-neutral-500">
-            DNS servers will be used in the order they are listed below.
+          <p :if={Enum.any?(@form.source.changes)} class="mb-4 text-neutral-500">
+            Upstream resolvers will be used by Clients in the order they are listed below.
+          </p>
+
+          <p :if={Enum.empty?(@form.source.changes)} class="text-neutral-500">
+            No upstream resolvers have been configured. Click <strong>New Resolver</strong>
+            to add one.
```
Comment (bmanifold) [inline elixir/apps/web/lib/web/live/settings/dns.ex]: I don't think this will work as expected.  If a DNS server has already been configured in a prior session this message will still be displayed, because there won't be any changes, but it will display the existing DNS server.

### Example:
Screenshot 2024-10-13 at 11 01 26 PM

File: elixir/apps/web/lib/web/live/actors/index.ex [hunk 3]
```diff
@@ -48,14 +48,20 @@ defmodule Web.Actors.Index do
     <.section>
       <:title><%= @page_title %></:title>
 
+      <:action>
+        <.docs_action path="/deploy/actors" />
```
Comment (bmanifold) [inline elixir/apps/web/lib/web/live/actors/index.ex]: I think this path needs to be `/deploy/users`.

**Comment by AndrewDryga** [comment]
@jamilbk sure we can remove it, in this PR I did not introduce the auto-complete, just tried to make it a little bit better. A lot of accounts are signed up using it but In theory for us increasing the sign-up friction is not the worst idea.

File: elixir/apps/web/lib/web/live/actors/index.ex (hunk 3 above)
Comment (AndrewDryga) [reply elixir/apps/web/lib/web/live/actors/index.ex]: Thanks!

**Comment by jamilbk** [comment]
> @jamilbk sure we can remove it, in this PR I did not introduce the auto-complete, just tried to make it a little bit better. A lot of accounts are signed up using it but In theory for us increasing the sign-up friction is not the worst idea.

Kk sgtm.

File: elixir/apps/web/lib/web/live/settings/dns.ex (hunk 2 above)
Comment (jamilbk) [reply elixir/apps/web/lib/web/live/settings/dns.ex]: Good catch! I wrote that pretty late 😅 should I push a quick fix?

Comment (AndrewDryga) [reply elixir/apps/web/lib/web/live/settings/dns.ex]: I already pushed a fix, no worries :)
